### PR TITLE
Renamings, Fs to check if siad.path exists, and some documentation

### PR DESCRIPTION
**Pull request by Mingling94**
- stop() wasn't calling its callback before
- start() now callbacks an error message when siad isn't found using fs
- clarified a bunch of comments
- DaemonManager is now SiadWrapper (call it how it is)
- Expanded README to have brief usage examples. Will need to solidify various features before devoting time to documenting more fully.

### CODE REVIEW

File: js/sia.js [hunk 1]
```diff
@@ -3,20 +3,19 @@
...
 
 /**
- * DaemonManager, a closure, initializes siad as a background process and
+ * Sia, a closure, initializes siad as a background process and
  * provides functions to interact with it
- * @class DaemonManager
+ * @class Sia
  */
-function DaemonManager () {
+function Sia () {
```
Comment (DavidVorick) [inline js/sia.js]: `Siad`?

File: js/sia.js [hunk 2]
```diff
@@ -62,10 +64,13 @@ function DaemonManager () {
   }
 
   // Checks whether siad is running on the current address
-  function checkRunning (callback) {
+  function checkIfSiadRunning (callback) {
```
Comment (DavidVorick) [inline js/sia.js:67]: :+1:

File: js/sia.js (hunk 1 above)
Comment (Mingling94) [inline js/sia.js]: But the node wrapper isn't really a `daemon`?

Comment (DavidVorick) [inline js/sia.js]: it's not really Sia either? It's supposed to wrap siad right? I am really tired, I might be missing something.

**Comment by Mingling94** [comment]
@lukechampine @DavidVorick boop

**Pull request merged**